sysvabi64/sysvabi64.rst

```diff
@@ -916,10 +923,639 @@ These follow the AAELF64_ base definition from Arm.
+independence and shareability of the program's text. Executable files
+and shared objects have their own separate procedure linkage
+tables. The same procedure linkage table format is used for both
+executables and shared objects.
+
+Following the steps below, the dynamic linker and the program
+cooperate to resolve symbolic references to functions through the
+procedure linkage table and the global offset table.
+
+ * When first creating the memory image of the program, the dynamic
+   linker sets the second and the third entries in the global offset
+   table to special values. The steps below explain more about these
+   values.
+
+ * Each shared object file in the process image has its own PLT, and
+   control transfers to a PLT entry only from within the same ELF
+   module.
+
+ * For illustration, assume the program calls ``name1``, which transfers
+   control to the Nth PLT entry in ``PLT[N]``.
+
+ * The ``PLT[N]`` entry loads the address of the GOT entry for
+   ``name1`` (``.got.plt[N]``) into register ``ip0``, and then loads
+   and jumps to the address held in that table entry using a ``BR``
+   instruction. Initially the GOT entry holds the address of the
+   ``PLT[0] entry``, the first entry in the PLT, not the real address
+   of ``name1``.
+
+ * Now the ``PLT[0]`` entry pushes the ``ip0`` and ``lr`` registers
+   onto the stack.  The dynamic linker will later use the stacked
+   value of ``ip0`` to compute a relocation index by subtracting the
+   base of the ``GOT`` The index will be used to select an entry from
+   the table of relocation entries addressed by the ``DT_JMPREL``
+   dynamic section entry. The designated relocation entry must have
+   type ``R_AARCH64_JUMP_SLOT``, and its offset will specify the
+   global offset table entry used by the branch instruction in
+   ``PLT[N]``, and also a symbol table index that references the
+   appropriate symbol, i.e. ``name1`` in this example.
+
+ * The ``PLT[0]`` entry then loads the adress of the third GOT entry
+   (``.got.plt[2]``) into ``ip0``, and the loads and jumps to the
+   address in the third table entry, which transfers control to the
+   dynamic linker.
+
+ * The value in ``ip0`` is used by the dynamic linker's entrypoint to
+   load the second GOT entry (``.got.plt[2]``) using ``[ip0,#-8]``,
+   which gives it one word of private information with which to
+   identify the calling module.
+
+ * The dynamic linker unwinds the stack, looks at the designated
+   relocation entry, finds the symbol's value, stores the actual
+   address of ``name1`` in ``.got.plt[N]`` , then transfers control to
+   the desired destination.  Subsequent executions of ``PLT[N]`` will
+   transfer control directly to ``name1``, without calling the dynamic
+   linker a second time. That is the ``BR`` instruction in ``PLT[N]``
+   will transfer to ``name1``, instead of ``PLT[0]``.
+
+The ``LD_BIND_NOW`` environment variable can change the dynamic linking
```

Comment:
And `DT_BIND_NOW`.

Full RELRO has been enabled for all packages in Fedora 23 and later.

Reply:
Ack, will add.